pytorch_lightning/trainer/connectors/signal_connector.py

```diff
@@ -0,0 +1,121 @@
+import logging
+import os
+import signal
+from signal import Signals
+from subprocess import call
+from typing import Callable, List, Optional, Union
+
+import _signal
+
+from pytorch_lightning.utilities.imports import _fault_tolerant_training
+
+log = logging.getLogger(__name__)
+
+
+class HandlersCompose:
+    def __init__(self, signal_handlers: Union[List[Callable], Callable]):
+        if not isinstance(signal_handlers, list):
+            signal_handlers = [signal_handlers]
+        self.signal_handlers = signal_handlers
+
+    def __call__(self, signum, frame):
+        for signal_handler in self.signal_handlers:
+            signal_handler(signum, frame)
+
+
+class SignalConnector:
+    def __init__(self, trainer, sigusr1_handler: Optional[Callable] = None, sigterm_handler: Optional[Callable] = None):
+        self.trainer = trainer
+        self.trainer._terminate_gracefully = False
+        self._sigusr1_handler = sigusr1_handler
+        self._sigterm_handler = sigterm_handler
+
+    @property
+    def sigusr1_handler(self) -> Optional[Callable]:
+        return self._sigusr1_handler
+
+    @sigusr1_handler.setter
+    def sigusr1_handler(self, sigusr1_handler: Callable) -> None:
+        self._sigusr1_handler = sigusr1_handler
+
+    @property
+    def sigterm_handler(self) -> Optional[Callable]:
+        return self._sigterm_handler
+
+    @sigterm_handler.setter
+    def sigterm_handler(self, sigterm_handler: Callable) -> None:
+        self._sigterm_handler = sigterm_handler
```

Comment:
What's the point of these getter/setters?

You can instead make the property public in the init.

Reply:
This isn't connected yet to the Accelerator. Advanced users might want to set them using a callback instead.

Reply:
But if the user set them outside of Lightning, they won't be overridden, so it would be fine.

Reply:
You are right, with the detection, it won't be required anymore.